### DISCUSSION TITLE
Onedrive connection

### DISCUSSION THREAD

**Discussion by Nounours13** [Studio-42/elFinder#3374] [Q&A]
Hi,
I attempt to connect elfinder (2.1.59) to onedrive.

I've set in connector.minimal.php this parameters:

define('ELFINDER_CONNECTOR_URL', 'http://localhost/testelfinder/php/connector.minimal.php');
define('ELFINDER_ONEDRIVE_CLIENTID','da67ae7f-aa11-xxxxxx'); //Key generated to Microsoft Azure
define('ELFINDER_ONEDRIVE_CLIENTSECRET','SGaSj9zzi6Uc683xxxxxxx'); //Key generated to Microsoft Azure

I've added this redirected URI in microsoft Azure
http://localhost/testelfinder/php/connector.minimal.php/netmount/onedrive/1

Then when i click on "mount network volume"

![image](https://user-images.githubusercontent.com/12816315/124629524-d2941680-de81-11eb-99c4-597d6bb1729d.png)

An other tab open whith autentification and this tab close and after on elfinder i've the same windows

![image](https://user-images.githubusercontent.com/12816315/124629757-05d6a580-de82-11eb-8e6f-df14725a45ae.png)

I can only click on "Goto Onedrive & approve" or "cancel". If i click to "Goto Onedrive & approve" there is the same thing and if i click to "cancel" the window close. and the drive not mount.

I've this informations on debug

![image](https://user-images.githubusercontent.com/12816315/124629930-3585ad80-de82-11eb-8d66-68ba3514437b.png)

Anybody can help me.

Thanks

**Comment by SysOpGeek** [comment]
I am also having this problem.  I am running it on a live server, and I have the Google Drive connector working.  Been banging my head against the keyboard the last few hours..

my file dir is 
https://......./files/

my  uri or redirect is set as.. at Microsoft/Azure
https://.......files/php/connector.minimal.php/netmount/onedrive/1

the file is there and returning the output of:
{"error":["errNetMount","1","errReauthRequire"]}

it was redirecting me to the authorization page and saying the wrong uri was set.
"Message: AADSTS50011: The reply URL specified in the request does not match the reply URLs configured for the application: 'xxxxxxxxx'."
I have the it set in the connector file also..

Anything anyone can tell me to keep some of my hair intact would be great... LOL... Thanks.

**Comment by nao-pon** [comment]
The "GotoOnedrive & approve" button contains the redirect URI in the script tag. Please check if the URL and the URL registered in Azure are the same.

![210707-135721](https://user-images.githubusercontent.com/1412630/124703122-abdeea00-df2c-11eb-8d08-8155614f2a71.png)

**Reply by Nounours13** [reply]
Hi Nao-Pon,
I've checked and is the same.

![image](https://user-images.githubusercontent.com/12816315/124884787-e8652100-dfd2-11eb-8dc8-f52a4f6b5eb7.png)

I've always the problem.

**Reply by Nounours13** [reply]
Hi Nao-Pon,
I've tested on your demo site my connection, and when i click to connect network drive, i can choose onedrive, and when i click to "Goto Onedrive & approve" there is a tab to login on my onedrive and when i select my credentials, the drive mount for an hour.
How to do this whitout create client_id and clientsecret on connector.php.
Thanks for your help.

**Reply by nao-pon** [reply]
All you need to do is set the following settings in your PHP connector and the correct redirect URI in Azure. If they are correct, you should be able to use OneDrive with your network mount.

```php
elFinder::$netDrivers['onedrive'] = 'OneDrive';
define('ELFINDER_ONEDRIVE_CLIENTID', 'Application (client) ID');
define('ELFINDER_ONEDRIVE_CLIENTSECRET', 'Your Client secret');
```

**Reply by Nounours13** [reply]
Hi nao-pon
Thanks for your anwer, but i don't understand what i've did wrong.
I've search in all debug mode. I've no error in all page but the drive not connect.
I've the same symptom.

That I don't underestand also, that how on your demo site I can connect my onedrive drive without problem while in your configuration.php you haven't client_id nor client_secret.

It's a mystery for me.

Cordially ...

**Comment by SysOpGeek** [comment]
I have verified the links are correct, now azure is returning there is a
trailing /

from (elfinder)
redirect_uri=https%3A%2F%2Fyes.mypc.rocks%2Fdesktop%2Ffiles%2Fphp%2Fconnector.minimal.php%2Fnetmount%2Fonedrive%2F1

from(azure setup)
https://yes.mypc.rocks/desktop/files/php/connector.minimal.php/netmount/onedrive/1
and
https://yes.mypc.rocks/desktop/files/php/connector.minimal.php/netmount/onedrive/1/

I have both in azure, have tried each separate and several other
combinations

**Comment by SysOpGeek** [comment]
Tried reinstalling, now i am seeing

errAccess invalid_client: AADSTS7000215: Invalid client secret is provided.
Trace ID: 989e0706-cc3c-407a-a237-62582a4bb802 Correlation ID:
71a01c1e-4253-41de-b5ee-610541f10e6a Timestamp: 2021-07-07 06:06:17Z

**Comment by SysOpGeek** [comment]
I got it....  I was using the Client ID, from Azure when I needed to be
using the Value....

**Reply by Nounours13** [reply]
I SysOpGeek
can you send me your configuration.minimal.php
Because i think i've a problem with mine. Thanks

**Reply by SysOpGeek** [reply]
Nounours13,  
Do you have a working configuration (without OneDrive)?  If so, are you familiar with composer?
I am currently working on writing step by step instructions for each connector...  Will be happy to help you get it running.  As for a copy of my connector file, it is setup with all my needs.  I would be happy to help you configure your own with the correct information, and debug any problems.   

I believe in sharing our knowledge...  Thats how we all  Learn... : )

**Reply by Nounours13** [reply]
Hi SysOpGeek and thanks for your answer.
So sorry but i don't use composer.
I've spend lot of time for try to connect to one drive, and i can't always connect to.
I've found the first problem on debug log. It's CURL.
The extension was enabled but don't start (on windows Uwamp server). I've check on Linux server and now I've an orther problem.
I've an error message:
_Undefined property: stdClass::$expires_in in elFinderVolumeOneDrive.class.php line 188_

I search where is the problem.
If you have the solution, I'am very happy.
In all case, when i've found the procedure to connect onedrive to elfinder, i send the correct procedure to connect one drive to elfinder.

cordially ../

**Reply by Nounours13** [reply]
Hi SysOpGeek , I've installed the last version of elfinder and now 've no erro on debug.
But the one drive drive is always not connected.

When i click on "Goto Onedrive & approve" an other tab open whith autentification and this tab close and after on elfinder i've the same windows. I'am still crazy ...

**Comment by SysOpGeek** [comment]
Are you in the US, would you be able to call Me?  I am currently without a
computer, however I am willing to help.

Thanks.

**Reply by Nounours13** [reply]
Hi SysOpGeek, 
I'm not in US, I'm in France.
If you want, i can send my connector.minimal.php.

Thanks for your help.

Cordially ...

**Reply by Nounours13** [reply]
Hi SysOpGeek,
I've "resolved" the error _Undefined property: stdClass::$expires_in in elFinderVolumeOneDrive.class.php line 188_.
I've installed the new version of elfinder 2.1.59 (2.1-src Nightly: db59327) and now there is no error.
However onedrive drive always doesn't connect.

I've the same symptom.
